pkg/user/user_integration_test.go

```diff
@@ -65,13 +65,53 @@ func TestUserHandler(t *testing.T) {
 		var user2 model.User
 		client.PostJSON(t, "/users", strings.NewReader(`{
 			"email":    "user2@dhis2.org",
-			"password": "oneoneoneoneone1"
+			"password": "oneoneoneoneoneoneone111"
 		}`), &user2)
 
 		require.Equal(t, "user2@dhis2.org", user2.Email)
 		require.Empty(t, user2.Password)
 	}
 
+	t.Run("SignUpFailure", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("InvalidEmail", func(t *testing.T) {
+			requestBody := strings.NewReader(`{
+				"email":    "not-a-valid-email",
+				"password": "oneoneoneoneoneoneone111"
+			}`)
+			response := client.Do(t, http.MethodPost, "/users", requestBody, http.StatusBadRequest, inttest.WithHeader("Content-Type", "application/json"))
+			require.Equal(t, "invalid email provided: not-a-valid-email", string(response))
+		})
+
+		t.Run("TooShortPassword", func(t *testing.T) {
+			requestBody := strings.NewReader(`{
+				"email":    "some@email.com",
+				"password": "short-password"
+			}`)
+			response := client.Do(t, http.MethodPost, "/users", requestBody, http.StatusBadRequest, inttest.WithHeader("Content-Type", "application/json"))
+			require.Equal(t, "password must be between 24 and 128 characters", string(response))
+		})
+
+		t.Run("TooLongPassword", func(t *testing.T) {
+			requestBody := strings.NewReader(`{
+				"email":    "some@email.com",
+				"password": "long-password-long-password-long-password-long-password-long-password-long-password-long-password-long-password-long-password-long-password"
```

Comment:
You could use `strings.Repeat("a", 129)` which makes it easy to spot why the password is invalid.